builtin/providers/aws/resource_aws_efs_file_system.go

```diff
@@ -149,15 +149,34 @@ func resourceAwsEfsFileSystemRead(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("EFS file system %q could not be found.", d.Id())
 	}
 
-	tagsResp, err := conn.DescribeTags(&efs.DescribeTagsInput{
-		FileSystemId: aws.String(d.Id()),
-	})
-	if err != nil {
-		return fmt.Errorf("Error retrieving EC2 tags for EFS file system (%q): %s",
-			d.Id(), err.Error())
+	tags := make([]*efs.Tag, 0)
+	var marker string
+	for {
+		params := &efs.DescribeTagsInput{
+			FileSystemId: aws.String(d.Id()),
+		}
+		if marker != "" {
+			params.Marker = aws.String(marker)
+		}
+
+		tagsResp, err := conn.DescribeTags(params)
+		if err != nil {
+			return fmt.Errorf("Error retrieving EC2 tags for EFS file system (%q): %s",
+				d.Id(), err.Error())
+		}
+
+		for _, tag := range tagsResp.Tags {
+			tags = append(tags, tag)
```

Comment:
instead of a `range` here I believe you can just directly append, since `tags` is `[]*efs.Tag`:

    tags = append(tags, tagsResp.Tags...)